### PR TITLE
Jira/PE-45: Integrate provisiond service with configuration manager

### PR DESCRIPTION
**Pull request by freddy61025**
https://issues.opennms.org/browse/PE-45

### CODE REVIEW

File: features/config/service/impl/src/main/java/org/opennms/features/config/service/impl/AbstractCmJaxbConfigDao.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        LOG.info("Loaded {} in {} ms", getDescription(), (endTime - startTime));
+
+        // If the config is not load in the first time, we will trigger the callbacks for this change
+        ENTITY_CLASS lastKnownEntity = lastKnownEntityMap.get(configId);
+        if (lastKnownEntity != null) {
+            synchronized (onReloadCausedChangeCallbacks) {
+                if (!onReloadCausedChangeCallbacks.isEmpty()) {
+                    LOG.debug("Calling onReloaded callbacks");
+                    try {
+                        onReloadCausedChangeCallbacks.forEach(c -> c.accept(config));
```
Comment (patrick-schweizer) [inline features/config/service/impl/src/main/java/org/opennms/features/config/service/impl/AbstractCmJaxbConfigDao.java:114]: the loop stops at the first failure. I wonder if we shouldn't try each callback even after one failed?

File: features/config/service/impl/src/main/java/org/opennms/features/config/service/impl/AbstractCmJaxbConfigDao.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        return config;
+    }
+
+    /**
+     * get default config object
+     *
+     * @return config object
+     */
+    public ENTITY_CLASS getConfig(String configId) throws IOException {
+        ENTITY_CLASS lastKnownEntity = lastKnownEntityMap.get(configId);
```
Comment (patrick-schweizer) [inline features/config/service/impl/src/main/java/org/opennms/features/config/service/impl/AbstractCmJaxbConfigDao.java:131]: nit: could be replaced by:
return lastKnownEntityMap.computeIfAbsent(configId, this::loadConfig);

File: opennms-config-tester/src/main/resources/META-INF/opennms/applicationContext-configTester.xml [hunk 3]
```diff
@@ -350,7 +350,7 @@
   <bean id="prometheusDataCollectionConfigDao" class="org.opennms.netmgt.dao.prometheus.PrometheusDataCollectionConfigDaoJaxb" />
 
   <bean id="provisiondConfigurationDao" class="org.opennms.netmgt.dao.jaxb.DefaultProvisiondConfigurationDao">
-    <property name="configResource" value="file:${opennms.home}/etc/provisiond-configuration.xml"/>
+    <!--property name="configResource" value="file:${opennms.home}/etc/provisiond-configuration.xml"/-->
```
Comment (patrick-schweizer) [inline opennms-config-tester/src/main/resources/META-INF/opennms/applicationContext-configTester.xml:353]: remove?

File: opennms-dao/src/main/java/org/opennms/netmgt/dao/jaxb/DefaultProvisiondConfigurationDao.java [hunk 4]
```diff
@@ -43,29 +43,27 @@
  * @author <a href="mailto:david@opennms.org">David Hustace</a>
  * @version $Id: $
  */
-public class DefaultProvisiondConfigurationDao extends AbstractJaxbConfigDao<ProvisiondConfiguration, ProvisiondConfiguration> implements ProvisiondConfigurationDao {
+public class DefaultProvisiondConfigurationDao extends AbstractCmJaxbConfigDao<ProvisiondConfiguration> implements ProvisiondConfigurationDao {
+
+    private final String configName = "Provisiond";
```
Comment (patrick-schweizer) [inline opennms-dao/src/main/java/org/opennms/netmgt/dao/jaxb/DefaultProvisiondConfigurationDao.java:48]: no constant?

**Review by patrick-schweizer — APPROVED** [review]
Minor stuff

File: features/config/service/impl/src/main/java/org/opennms/features/config/service/impl/AbstractCmJaxbConfigDao.java (hunk 2 above)
Comment (freddy61025) [reply features/config/service/impl/src/main/java/org/opennms/features/config/service/impl/AbstractCmJaxbConfigDao.java:131]: thanks

File: opennms-config-tester/src/main/resources/META-INF/opennms/applicationContext-configTester.xml (hunk 3 above)
Comment (freddy61025) [reply opennms-config-tester/src/main/resources/META-INF/opennms/applicationContext-configTester.xml:353]: removed

File: features/config/service/impl/src/main/java/org/opennms/features/config/service/impl/AbstractCmJaxbConfigDao.java (hunk 1 above)
Comment (freddy61025) [reply features/config/service/impl/src/main/java/org/opennms/features/config/service/impl/AbstractCmJaxbConfigDao.java:120]: added TODO comment to consider when working on PE-13

File: opennms-dao/src/main/java/org/opennms/netmgt/dao/jaxb/DefaultProvisiondConfigurationDao.java (hunk 4 above)
Comment (freddy61025) [reply opennms-dao/src/main/java/org/opennms/netmgt/dao/jaxb/DefaultProvisiondConfigurationDao.java]: fixed

**Review by patrick-schweizer — DISMISSED** [review]

**Review by patrick-schweizer — APPROVED** [review]

**Pull request merged**